testscripts/testrunner/examplesrunner.go

```diff
@@ -1,82 +1,143 @@
 package testrunner
 
 import (
+	"fmt"
+	"io/fs"
 	"os"
 	"os/exec"
 	"path/filepath"
+	"strings"
 	"testing"
 
 	"github.com/pkg/errors"
 	"github.com/rogpeppe/go-internal/testscript"
 	"go.jetpack.io/devbox/internal/debug"
+	"go.jetpack.io/devbox/internal/impl"
 )
 
 // RunExamplesTestscripts generates testscripts for each example devbox-project.
-// For now, we prototype with the "go" example project. TODO savil: generalize.
 func RunExamplesTestscripts(t *testing.T, examplesDir string) {
 	wd, err := os.Getwd()
 	if err != nil {
 		t.Error(err)
 	}
 
-	// TODO savil. Change this to handle _all_ of the example folder's devbox-projects
 	examplesDir = filepath.Join(wd, examplesDir)
-	projectDir := filepath.Join(examplesDir, "development", "go", "hello-world")
-	runSingleExampleTestscript(t, examplesDir, projectDir)
-}
+	err = filepath.WalkDir(examplesDir, func(path string, entry os.DirEntry, err error) error {
+		if err != nil {
+			return err
+		}
 
-func runSingleExampleTestscript(t *testing.T, examplesDir, projectDir string) {
-	testscriptDir, err := generateTestscript()
+		if !entry.IsDir() {
+			return nil
+		}
+
+		// skip directories that do not have a devbox.json defined
+		configPath := filepath.Join(path, "devbox.json")
+		if _, err := os.Stat(configPath); err != nil {
+			if errors.Is(err, fs.ErrNotExist) {
+				return nil
+			}
+			return err
+		}
+
+		// skip configs that do not have a run_test defined
+		config, err := impl.ReadConfig(configPath)
+		if err != nil {
+			return err
+		}
+		if _, ok := config.Shell.Scripts["run_test"]; !ok {
+			t.Logf("skipping config due to missing run_test at: %s\n", path)
+			return nil
+		}
+
+		// TODO savil. Resolve these.
+		skipList := []string{
```

Comment:
If you define this as a `map[string]bool` outside of the walk function then you can simplify down to:

```go
if shouldSkip[path] {
	t.Logf("skipping example %q because it's broken", path)
	return nil
}
```